libraries/AP_Mission/AP_Mission_Commands.cpp

```diff
@@ -0,0 +1,112 @@
+#include "AP_Mission.h"
+
+#include <GCS_MAVLink/GCS.h>
+#include <AP_Gripper/AP_Gripper.h>
+#include <AP_ServoRelayEvents/AP_ServoRelayEvents.h>
+
+bool AP_Mission::start_command_do_gripper(const AP_Mission::Mission_Command& cmd)
+{
+    AP_Gripper *gripper = AP::gripper();
+    if (gripper == nullptr) {
+        return true;
+    }
+
+    // Note: we ignore the gripper num parameter because we only
+    // support one gripper
+    switch (cmd.content.gripper.action) {
+    case GRIPPER_ACTION_RELEASE:
+        gripper->release();
+        // Log_Write_Event(DATA_GRIPPER_RELEASE);
```

Comment:
I haven't done a lot with these as I don't do much with copter, but why not make `Log_Write_Event` a required/generic DataFlash method/type? Seems like it would allow stuff to move out to libraries better, and I think (but haven't checked) that some of the existing events would be easier to track/generate in the library code then Copter vehicle code as it stands?